### ISSUE TITLE
error in streaming video, IOS (_duration is null)

### ISSUE THREAD

**Issue by numericni** [brianegan/chewie#360]
Error show in IOS for streaming video:

The following NoSuchMethodError was thrown while handling a gesture:
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration

Option:

_chewieController = ChewieController(
      videoPlayerController: _videoPlayerController1,
      aspectRatio: 3 / 2,
      autoPlay: false,
      looping: true,
      isLive: true,

Regards

**Comment by hedrondantas** [comment]
Same error

**Comment by hedrondantas** [comment]
sent tap down
════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration
When the exception was thrown, this was the stack:
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
#1      Duration.>=  (dart:core/duration.dart:173:63)
#2      _MaterialControlsState._playPause 
package:chewie/src/material_controls.dart:326
#3      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:184
#4      TapGestureRecognizer.handleTapUp 
package:flutter/…/gestures/tap.dart:524
#5      BaseTapGestureRecognizer._checkUp 
package:flutter/…/gestures/tap.dart:284
#6      BaseTapGestureRecognizer.acceptGesture 
package:flutter/…/gestures/tap.dart:256
#7      GestureArenaManager.sweep 
package:flutter/…/gestures/arena.dart:158
#8      GestureBinding.handleEvent 
package:flutter/…/gestures/binding.dart:224
#9      GestureBinding.dispatchEvent 
package:flutter/…/gestures/binding.dart:200
#10     GestureBinding._handlePointerEvent 
package:flutter/…/gestures/binding.dart:158
#11     GestureBinding._flushPointerEventQueue 
package:flutter/…/gestures/binding.dart:104
#12     GestureBinding._handlePointerDataPacket 
package:flutter/…/gestures/binding.dart:88
#16     _invoke1  (dart:ui/hooks.dart:267:10)
#17     _dispatchPointerDataPacket  (dart:ui/hooks.dart:176:5)
(elided 3 frames from dart:async)
Handler: "onTap"
Recognizer:
  TapGestureRecognizer#1671b
════════════════════════════════════════════════════════════════════════════════════════════════════
Lost connection to device.
Exited (sigterm)

**Comment by hedrondantas** [comment]
`    sent tap down
════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration
When the exception was thrown, this was the stack:
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
#1      Duration.>=  (dart:core/duration.dart:173:63)
#2      _MaterialControlsState._playPause 
package:chewie/src/material_controls.dart:326
#3      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:184
#4      TapGestureRecognizer.handleTapUp 
package:flutter/…/gestures/tap.dart:524
#5      BaseTapGestureRecognizer._checkUp 
package:flutter/…/gestures/tap.dart:284
#6      BaseTapGestureRecognizer.acceptGesture 
package:flutter/…/gestures/tap.dart:256
#7      GestureArenaManager.sweep 
package:flutter/…/gestures/arena.dart:158
#8      GestureBinding.handleEvent 
package:flutter/…/gestures/binding.dart:224
#9      GestureBinding.dispatchEvent 
package:flutter/…/gestures/binding.dart:200
#10     GestureBinding._handlePointerEvent 
package:flutter/…/gestures/binding.dart:158
#11     GestureBinding._flushPointerEventQueue 
package:flutter/…/gestures/binding.dart:104
#12     GestureBinding._handlePointerDataPacket 
package:flutter/…/gestures/binding.dart:88
#16     _invoke1  (dart:ui/hooks.dart:267:10)
#17     _dispatchPointerDataPacket  (dart:ui/hooks.dart:176:5)
(elided 3 frames from dart:async)
Handler: "onTap"
Recognizer:
  TapGestureRecognizer#1671b
════════════════════════════════════════════════════════════════════════════════════════════════════
Lost connection to device.
Exited (sigterm)
`

**Comment by hedrondantas** [comment]
════════ Exception caught by gesture ═══════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration

When the exception was thrown, this was the stack
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
#1      Duration.>=  (dart:core/duration.dart:173:63)
#2      _MaterialControlsState._playPause 
package:chewie/src/material_controls.dart:326
#3      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:184
#4      TapGestureRecognizer.handleTapUp 
package:flutter/…/gestures/tap.dart:524
...
Handler: "onTap"
Recognizer: TapGestureRecognizer#1671b
    debugOwner: GestureDetector
    state: ready
    won arena
    finalPosition: Offset(27.0, 441.5)
    finalLocalPosition: Offset(27.0, 22.5)
    button: 1
    sent tap down
════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY GESTURE ╞═══════════════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration
When the exception was thrown, this was the stack:
#0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
#1      Duration.>=  (dart:core/duration.dart:173:63)
#2      _MaterialControlsState._playPause 
package:chewie/src/material_controls.dart:326
#3      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:184
#4      TapGestureRecognizer.handleTapUp 
package:flutter/…/gestures/tap.dart:524
#5      BaseTapGestureRecognizer._checkUp 
package:flutter/…/gestures/tap.dart:284
#6      BaseTapGestureRecognizer.acceptGesture 
package:flutter/…/gestures/tap.dart:256
#7      GestureArenaManager.sweep 
package:flutter/…/gestures/arena.dart:158
#8      GestureBinding.handleEvent 
package:flutter/…/gestures/binding.dart:224
#9      GestureBinding.dispatchEvent 
package:flutter/…/gestures/binding.dart:200
#10     GestureBinding._handlePointerEvent 
package:flutter/…/gestures/binding.dart:158
#11     GestureBinding._flushPointerEventQueue 
package:flutter/…/gestures/binding.dart:104
#12     GestureBinding._handlePointerDataPacket 
package:flutter/…/gestures/binding.dart:88
#16     _invoke1  (dart:ui/hooks.dart:267:10)
#17     _dispatchPointerDataPacket  (dart:ui/hooks.dart:176:5)
(elided 3 frames from dart:async)
Handler: "onTap"
Recognizer:
  TapGestureRecognizer#1671b
════════════════════════════════════════════════════════════════════════════════════════════════════
Lost connection to device.
Exited (sigterm)

**Comment by hedrondantas** [comment]
Only happens in live streams.

**Comment by numericni** [comment]
> Only happens with live streams.

can you solve the problem?

**Comment by hedrondantas** [comment]
> 
> 
> > Only happens with live streams.
> 
> can you solve the problem?

Just downgrade the video_player plugin, it's a bug in the lastest version.

**Comment by numericni** [comment]
> > > Only happens with live streams.
> > 
> > 
> > can you solve the problem?
> 
> Just downgrade the video_player plugin, it's a bug in the lastest version.

Which version worked for you?

**Comment by HyperJames** [comment]
I've checked this on the most recent versions of Chewie (0.12.0) and video_player (1.0.1) and the problem persists on iOS 14.2. 
I've then **downgraded** to Chewie 0.9.10 and video_player 0.10.12+5 (still on iOS 14.2) and the issue is not present. So the bug has been introduced between Chewie 0.9.10 and 0.12.0.

**Comment by mahmoudalaa97** [comment]
Any update for this issue i have same issue 

**Plugin Version:**
>  chewie: ^0.12.0
>  video_player: ^1.0.1

**Flutter doctor:**
```
√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.19041.630], locale en-GB)
    • Flutter version 1.22.4 at I:\flutter\flutter
    • Framework revision 1aafb3a8b9 (3 weeks ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4


[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at G:\android\Android\sdkn
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 4.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 49.0.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.51.1)
    • VS Code at C:\Users\Java\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.16.0

[√] Connected device (4 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86    • Android 7.0 (API 24) (emulator)
    • Web Server (web)                   • web-server    • web-javascript • Flutter Tools
    • Chrome (web)                       • chrome        • web-javascript • Google Chrome 87.0.4280.88
    • Edge (web)                         • edge          • web-javascript • Microsoft Edge 87.0.664.47

• No issues found!
```

**Error:**
```
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The getter '_duration' was called on null.
Receiver: null
Tried calling: _duration
The relevant error-causing widget was: 
  Chewie file:/lib/src/widgets/video_widget.dart:70:12
════════════════════════════════════════════════════════════════════════════════════════════════════
```

**Comment by aissamouajib** [comment]
@numericni @mahmoudalaa97 having the same issue here, did you guys find any solution?

**Comment by numericni** [comment]
hola, para que funcione en IOS se tiene que comentar la linea 315 del archivo
#video_player-1.0.1/ios/Classes/FLTVideoPlayerPlugin.m

Estas 3 lineas tienes que comentar del archivo FLTVideoPlayerPlugin.m 
```
 if ([self duration] == 0) {
      return;
    }
```

Saludos

**Comment by nstrelow** [comment]
This should be fixed with 0.12.1.

Please check if it works now.

**Comment by nstrelow** [comment]
Should be fixed in 0.12.1
